File: filesystem/lib/copy/command.lua
function entrypoint()
    local from = source
    local to = destination
    if from == nil then
        error("No source path specified.")
    end
    if to == nil then
        error("No destination path specified.")
    end
    if not fs.exists(from) then
        error("Source path does not exist.")
    end
    if fs.exists(to) then
        error("Destination path already exists.")
    end

    fs.copy(from, to)
end

File: filesystem/lib/copy/command.py
from pathlib import Path
import shutil

import shfl



def entrypoint(ws, options, bg_mode):
    current_dir = Path(shfl.currentDirectory(ws))

    src = current_dir / options["source"]
    dest = current_dir / options["destination"]
    if not src.exists():
        print("Source path does not exist")
        return "SRC_NOT_EXIST"
    if not dest.is_dir():
        print("Destination path is not a directory")
        return "DEST_NOT_DIR"
    if not dest.exists():
        print("Destination path does not exist")
        return "DEST_NOT_EXIST"

    if src.is_dir():
        shutil.copytree(str(src), str(dest / src.name))
    else:
        shutil.copy(str(src), str(dest / src.name))

    return str(dest / src.name)